File: src/2020/2/python/2.1.py
with open("src/2020/2/inputs.txt", "r") as f:
    inputs = list(f.read().splitlines())
    counter = 0
    for occur in inputs:
        splitted = occur.split(" ")
        minimum = int(splitted[0].split("-")[0])
        maximum = int(splitted[0].split("-")[1])
        letter = splitted[1][0]
        chain = splitted[2]
        occur = len(chain.split(letter)) - 1
        if occur >= minimum and occur <= maximum:
            counter += 1
    print(counter)

File: src/2020/2/js/2.1.js
fs = require("fs");
fs.readFile("src/2020/2/inputs.txt", "utf8", function (err, data) {
  const inputs = data.split("\n");
  console.time();
  console.log(
    inputs.filter((input) => {
      const [, min, max, letter, chain] = input.split(/(\d+)-(\d+) (\w): (.+)/);
      const occur = chain.split(letter).length - 1;
      return occur >= min && occur <= max;
    }).length
  );
  console.timeEnd();
});
